app/Http/Controllers/BankIntegrationController.php

```diff
@@ -288,12 +336,20 @@ public function removeAccount(AdminBankIntegrationRequest $request, $acc_id)
      */
     public function getTransactions(AdminBankIntegrationRequest $request)
     {
-        /** @var \App\Models\User $user */
-        $user = auth()->user();
+        /** @var \App\Models\Account $account */
+        $account = auth()->user()->account();
 
-        $user->account->bank_integrations->each(function ($bank_integration) use ($user) {
-            (new ProcessBankTransactions($user->account->bank_integration_account_id, $bank_integration))->handle();
-        });
+        if (Ninja::isHosted() && $account->isPaid() && $account->plan == 'enterprise') {
+            $account->bank_integrations()->where('integration_type', BankIntegration::INTEGRATION_TYPE_YODLEE)->where('auto_sync', true)->where('is_deleted', false)->cursor()->each(function ($bank_integration) use ($account) {
```

Comment:
@turbo124 I added the where check is_deleted. should it even be required in all cases?